core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -775,6 +775,24 @@ class DynamicBrokerReconfigurationTest extends ZooKeeperTestHarness with SaslSet
     verifyMarkPartitionsForTruncation()
   }
 
+  @Test
+  def testDefaultValueRestoredAfterDeleteDynamicConfig(): Unit = {
```

Comment:
Can we move this into `DynamicBrokerReconfigurationTest.testDefaultTopicConfig()` and also verify that configs of existing topics have been updated? Or add a topic in this test to verify.